streaming/java/streaming-runtime/src/main/java/io/ray/streaming/runtime/worker/tasks/StreamTask.java

```diff
@@ -30,103 +26,115 @@
-      }
+      DataWriter writer = new DataWriter(channelIDs, outputActors, jobWorker.getWorkerConfig());
+      collectors.add(new OutputCollector(writer, channelIDs, outputActors.values(),
+          executionVertex.getOutputEdges().get(0).getPartition()));
     }
 
     // consumer
-    List<ExecutionEdge> inputEdges = executionNode.getInputsEdges();
+    List<ExecutionEdge> inputEdges = executionVertex.getInputEdges();
     Map<String, BaseActor> inputActors = new HashMap<>();
     for (ExecutionEdge edge : inputEdges) {
-      Map<Integer, BaseActor> taskId2Worker = executionGraph
-          .getTaskId2WorkerByNodeId(edge.getSrcNodeId());
-      taskId2Worker.forEach((srcTaskId, srcActor) -> {
-        String queueName = ChannelID.genIdStr(srcTaskId, taskId, executionGraph.getBuildTime());
-        inputActors.put(queueName, srcActor);
-      });
+      String queueName = ChannelID.genIdStr(
+          edge.getSourceExecutionVertex().getExecutionVertexId(),
+          taskId,
+          executionVertex.getBuildTime());
+      inputActors.put(queueName, edge.getSourceExecutionVertex().getWorkerActor());
     }
     if (!inputActors.isEmpty()) {
       List<String> channelIDs = new ArrayList<>();
       inputActors.forEach((k, v) -> {
         channelIDs.add(k);
       });
       LOG.info("Register queue consumer, queues {}.", channelIDs);
-      reader = new DataReader(channelIDs, inputActors, queueConf);
+      reader = new DataReader(channelIDs, inputActors, jobWorker.getWorkerConfig());
     }
 
-    RuntimeContext runtimeContext = new RayRuntimeContext(
-        worker.getExecutionTask(), worker.getConfig(), executionNode.getParallelism());
+    RuntimeContext runtimeContext = new StreamingRuntimeContext(executionVertex,
+        jobWorker.getWorkerConfig().configMap, executionVertex.getParallelism());
 
     processor.open(collectors, runtimeContext);
-
-    Runtime.getRuntime().addShutdownHook(new Thread(() -> {
-      try {
-        // Make DataReader stop read data when MockQueue destructor gets called to avoid crash
-        StreamTask.this.cancelTask();
-      } catch (Exception e) {
-        e.printStackTrace();
-      }
-    }));
+    LOG.debug("Finished preparing stream task.");
   }
 
+  /**
+   * Task initialization related work.
+   */
   protected abstract void init() throws Exception;
 
+  /**
+   * Stop running tasks.
+   */
   protected abstract void cancelTask() throws Exception;
 
   public void start() {
     this.thread.start();
-    LOG.info("started {}-{}", this.getClass().getSimpleName(), taskId);
+    LOG.info("Start stream task: {}-{}", this.getClass().getSimpleName(), taskId);
+  }
+
+  /**
+   * Close running tasks.
+   */
+  public void close() {
+    this.running = false;
+    if (thread.isAlive() && !Ray.getRuntimeContext().isSingleProcess()) {
+      //Runtime halt is proposed cause System.exist can't kill process absolutely.
```

Comment:
```suggestion
      // `Runtime.halt` is used because System.exist can't ensure kill process.
```

Reply:
`Runtime.halt` is used because System.exist can't ensure the process killing.

Reply:
Runtime.halt is used because System.exist can't ensure killing the process.

Reply:
Should be ensure sb. sth(or sth.).